compact.go

```diff
@@ -476,9 +486,14 @@ func (w *instrumentedChunkWriter) WriteChunks(chunks ...chunks.Meta) error {
 func (c *LeveledCompactor) write(dest string, meta *BlockMeta, blocks ...BlockReader) (err error) {
 	dir := filepath.Join(dest, meta.ULID.String())
 	tmp := dir + ".tmp"
-
+	var writers []io.Closer
+	var merr MultiError
 	defer func(t time.Time) {
-		if err != nil {
+		merr.Add(err)
+		for _, w := range writers {
+			merr.Add(w.Close())
+		}
+		if merr.Err() != nil {
 			c.metrics.failed.Inc()
 			// TODO(gouthamve): Handle error how?
 			if err := os.RemoveAll(tmp); err != nil {
```

Comment:
hm.. we don't want to remove tmp anyway?

Reply:
only when there is an error otherwise if everything was ok we don't have a tmp folder as this is renamed in `renameFile(tmp, dir);`

Reply:
I guess we can still run it as RemoveAll does nothing if the folder doesn't exist

Reply:
refactored